general/hip.hpp

```diff
@@ -15,16 +15,15 @@
 #include "../config/config.hpp"
 #include "error.hpp"
 
-#ifdef MFEM_USE_HIP
-#include <hip/hip_runtime.h>
-#endif
-
 // HIP block size used by MFEM.
 #define MFEM_HIP_BLOCKS 256
 
 #ifdef MFEM_USE_HIP
 #define MFEM_DEVICE __device__
+#define MFEM_LAMBDA __host__ __device__
```

Comment:
It seems inconsistent that the hip `MFEM_LAMBDA` is defined as `__host__ __device__` whereas the cuda one is defined as `__host__`, is there a reason for that?

Reply:
the two compilers behave differently